settings.py

```diff
@@ -208,31 +209,20 @@ def get_dosbox_scaler(self):
 
     def __assure_sf2_exists__(self):
         sf2 = self.__get_str__('midi', 'soundfont', DEFAULT_MIDI_SOUNDFONT)
-        sf2_search = [
-            ['/usr/share/sounds/sf2'],
-            ['/usr/share/soundfonts'],
-            ['/usr/local/share/sounds/sf2'],
-            ['/usr/local/share/soundfonts'],
-            [xdg.DATA_HOME, 'sounds/sf2'],
-            [xdg.DATA_HOME, 'soundfonts'],
-            [self.distdir, 'share/sounds/sf2'],
-            [self.distdir, 'share/soundfonts'],
-        ]
-        selected = ''
-        default = ''
-        os_default = ''
-        for path in sf2_search:
-            # TODO avoid unnecessary list append after Python 3.5 is dropped
-            selected_path = os.path.join(*(path + [sf2]))
-            default1_path = os.path.join(*(path + [DEFAULT_MIDI_SOUNDFONT]))
-            default2_path = os.path.join(*(path + ['default.sf2']))
-            if os.path.isfile(selected_path):
-                selected = selected_path
-            if os.path.isfile(default1_path):
-                default = default1_path
-            if os.path.isfile(default2_path):
-                os_default = default2_path
-        use_sf2 = selected or default or os_default
+        data_dirs = os.getenv('XDG_DATA_DIRS', '/usr/share').split(os.pathsep)
+        if xdg.DATA_HOME not in data_dirs:
+            data_dirs.insert(0, xdg.DATA_HOME)
+        data_dirs.insert(0, os.path.join(self.distdir, 'share'))
+        use_sf2 = ''
+        sf2_paths = (os.path.join(d, s, n) for n, d, s in itertools.product(
```

Comment:
This is so nice :)

Reply:
GitHub selected weird lines when I wanted specifically 217 - I was talking about using `itertools.product` here.